algoliasearch-core/src/test/java/com/algolia/search/JacksonParserTest.java

```diff
@@ -974,4 +977,21 @@ void testConsequencePromotion() throws IOException {
     assertThat(promote.get(1).getObjectID()).isNull();
     assertThat(promote.get(1).getObjectIDs()).isEqualTo(Arrays.asList("b", "c"));
   }
+
+  @Test
+  void rulesValidityTimeRange() throws IOException {
+    ZoneOffset offset = ZoneOffset.of("+05:30");
+    // Nanoseconds precisions will be lost and should not be considered.
+    TimeRange timerange =
+        new TimeRange(
+            OffsetDateTime.of(2021, 5, 1, 0, 0, 0, 0, offset),
+            OffsetDateTime.of(2021, 8, 30, 0, 23, 59, 0, offset));
+
+    String json = Defaults.getObjectMapper().writeValueAsString(timerange);
+    assertThat(json).isEqualTo("{\"from\":1619807400,\"until\":1630263239}");
+
+    TimeRange retrieveTimeRange = Defaults.getObjectMapper().readValue(json, TimeRange.class);
+    assertThat(timerange.getFrom().isEqual(retrieveTimeRange.getFrom())).isTrue();
```

Comment:
```suggestion
    assertThat(timerange.getFrom()).isEqual(retrieveTimeRange.getFrom());
```

should it be something more like this? it seems confusing to use isEqual & isTrue

Reply:
Fixed! the old version of our testing library assertj wasn't behaving as expected.
Updating it allows the use of a better syntax :)